sqlx-core/src/query_builder.rs

```diff
@@ -323,6 +329,10 @@ where
             separated.push_unseparated(")");
         }
 
+        if empty_checker {
+            tracing::warn!("Pushing empty values, QueryBuilder may not build correct sql query!");
+        }
```

Comment:
It would probably be helpful to elevate this to a panic in debug mode. Just use `debug_assert!(separated.push_separator, "<message>");`